src/CachePeer.cc

```diff
@@ -28,6 +29,8 @@ CachePeer::CachePeer(const char * const hostname):
 
 CachePeer::~CachePeer()
 {
+    fwdPconnPool->notePeerRemoved(this);
```

Comment:
We probably do not want to ask CachePeer destructor with iterating all possible asynchronous tasks/modules that may store this CachePeer.  This PR may be (mis)interpreted as moving closer to doing that unwanted architecture. I believe we need a different solution. We probably should start refcounting CachePeers. And we might even need some kind of CachePeer user registry (but I hope not).

Regardless of the above, this PR has changes that we are likely to eventually use. I am going to close this PR for now to reduce noise, but we will come back to this problem and will probably reuse some of this code.